web/libs/editor/src/lib/AudioUltra/Regions/Segment.ts

```diff
@@ -168,6 +170,21 @@ export class Segment extends Events<SegmentEvents> {
     return this.xStart + this.width;
   }
 
+  get yStart() {
+    const { timelinePlacement, timelineHeight } = this;
+    const timelineLayer = this.visualizer.getLayer("timeline");
+    const timelineTop = timelinePlacement === defaults.timelinePlacement;
+    const top = timelineLayer?.isVisible && timelineTop ? timelineHeight : 0;
+
+    return top;
+  }
+
+  get yEnd() {
+    const { height } = this.visualizer;
+    const { timelineHeight } = this;
+    return this.yStart + (height - timelineHeight);
+  }
```

Comment:
Thank you, this is a nice refactor that makes the readability of the code much better.